bouncer/src/repo/manipulator/modelconvertor/import/repo_model_import_3drepo.cpp

```diff
@@ -16,434 +16,1001 @@
+	}
+
+	virtual void StartObject() override {
+		material = repo::lib::repo_material_t::DefaultMaterial();
+		material.opacity = 0;
+		material.shininess = material.shininess / 5;
+		texture = -1;
+	}
+
+	virtual void EndObject() override {
+		material.opacity = 1 - material.opacity;
+		material.shininess = material.shininess * 5;
+		builder->createMaterial(material, texture);
+	}
+};
+
+struct GeometryParser : public ObjectParser
+{
+	GeometryRecord& geometry;
+
+	GeometryParser(GeometryRecord& geometry)
+		:geometry(geometry)
+	{
+		parsers["primitive"] = new NumberParser(geometry.primitive);
+		parsers["numIndices"] = new NumberParser(geometry.numIndices);
+		parsers["numVertices"] = new NumberParser(geometry.numVertices);
+		parsers["indices"] = new RangeParser(geometry.indices);
+		parsers["vertices"] = new RangeParser(geometry.vertices);
+		parsers["normals"] = new RangeParser(geometry.normals);
+		parsers["uv"] = new RangeParser(geometry.uv);
+		parsers["material"] = new NumberParser(geometry.material);
+	}
+
+	virtual void StartObject() {
+		geometry.hasGeometry = true;
+	}
+};
+
+struct NodeParser : public ObjectParser
+{
+	NodeRecord record;
+	Builder* builder;
+
+	NodeParser(Builder* builder)
+		:builder(builder)
+	{
+		parsers["id"] = new NumberParser(record.id);
+		parsers["name"] = new StringParser(record.name);
+		parsers["parent"] = new NumberParser(record.parentId);
+		parsers["metadata"] = new MetadataParser(record.metadata);
+		parsers["geometry"] = new GeometryParser(record.geometry);
+		parsers["bounds"] = new BoundsParser(record.bounds);
+		parsers["transformation"] = new RepoMatrixParser(record.matrix);
+	}
+
+	virtual void StartObject() {
+		record = NodeRecord();
+	}
+
+	virtual void EndObject() {
+		builder->createNode(record);
+	}
+};
+
+struct TreeParser : public ObjectParser
+{
+	TreeParser(Builder* builder)
+	{
+		parsers["nodes"] = new ArrayParser(new NodeParser(builder));
+		parsers["materials"] = new ArrayParser(new MaterialParser(builder));
+		parsers["textures"] = new ArrayParser(new TextureParser(builder));
+	}
+};
+
+/*
 * Will parse the entire BIM file and store the results in
 * temporary datastructures in preperation for scene generation.
 * @param filePath
 * @param err
 */
-bool RepoModelImport::importModel(std::string filePath, uint8_t& err)
+bool RepoModelImport::importModel(std::string filePath, std::shared_ptr<repo::core::handler::AbstractDatabaseHandler> handler, uint8_t& err)
 {
-	orgFile = filePath;
 	std::string fileName = getFileName(filePath);
 
 	repoInfo << "IMPORT [" << fileName << "]";
 	repoInfo << "=== IMPORTING MODEL WITH REPO IMPORTER ===";
 
 	// Reading in file
-	finCompressed = new std::ifstream(filePath, std::ios_base::in | std::ios::binary);
+	std::ifstream finCompressed(filePath, std::ios_base::in | std::ios::binary);
 	if (finCompressed)
 	{
-		inbuf = new boost::iostreams::filtering_streambuf<boost::iostreams::input>();
-#ifndef REPO_BOOST_NO_GZIP
```

Comment:
If we're making the assumption that bouncer always gets compiled with boost gzip library (which I guess is fair) then we need to remove the flag on CMAKE